src/tuesday/core/plotting/powerspectra.py

```diff
@@ -0,0 +1,389 @@
+    """
+    Plot a 2D power spectrum.
+
+    Parameters
+    ----------
+    wavemodes : un.Quantity
+        Wavemodes [kperp, kpar].
+        Accepted units are 1/Mpc or h/Mpc.
+    power_spectrum : un.Quantity
+        Power spectrum array of shape [Nsamples, Nkperp, Nkpar]
+        or [Nkperp, Nkpar].
+        There are six accepted units: mK^2 Mpc^3, mK^2 Mpc^3/h^3, mK^2,
+        or dimensionless instead of mK^2.
+    fig : plt.Figure, optional
+        Figure object to plot on. If None, a new figure is created.
+    axs : plt.Axes | list[plt.Axes], optional
+        Axes object(s) to plot on. If None, new axes are created.
+    title : str, optional
+        Title(s) of the plot.
+    xlabel : str, optional
+        Label for the x-axis.
+    ylabel : str, optional
+        Label for the y-axis.
+    clabel : str, optional
+        Label for the colorbar.
+    cmap : str, optional
+        Colormap for the plot.
+    fontsize : float, optional
+        Font size for the plot labels.
+    vmin : float, optional
+        Minimum value for the color scale.
+    vmax : float, optional
+        Maximum value for the color scale.
+    log : list[bool], optional
+        List of booleans to set the kperp, kpar, and PS axes to log scale.
+    smooth : float, optional
+        Standard deviation for Gaussian smoothing.
+        If True, uses a standard deviation of 1.
+    """
+    rcParams.update({"font.size": fontsize})
+    kperp = wavemodes[0]
+    kpar = wavemodes[1]
+    if xlabel is None:
+        if kperp.unit == 1 / un.Mpc:
+            xlabel = r"$k_\perp \, [\rm{Mpc}^{-1}]$"
+        elif kperp.unit == littleh / un.Mpc:
+            xlabel = r"$k_\perp \, [h \, \rm{Mpc}^{-1}]$"
+        else:
+            raise ValueError("kperp must be in units of 1/Mpc or h/Mpc.")
+
+    if ylabel is None:
+        if kpar.unit == 1 / un.Mpc:
+            ylabel = r"$k_\parallel \, [\rm{Mpc}^{-1}]$"
+        elif kpar.unit == littleh / un.Mpc:
+            ylabel = r"$k_\parallel \, [h \, \rm{Mpc}^{-1}]$"
+        else:
+            raise ValueError("kpar must be in units of 1/Mpc or h/Mpc.")
+
+    if clabel is None:
+        if power_spectrum.unit == un.mK**2 * un.Mpc**3:
+            clabel = r"$P(k) \, [\rm{mK}^2 \, \rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.mK**2 * un.Mpc**3 / littleh**3:
+            clabel = r"$P(k) \, [\rm{mK}^2 \, h^{-3} \, \rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.mK**2:
+            clabel = r"$\Delta^2_{21} \, [\rm{mK}^2]$"
+        elif power_spectrum.unit == un.Mpc**3:
+            clabel = r"$P(k) \, [\rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.Mpc**3 / littleh**3:
+            clabel = r"$P(k) \, [h^{-3} \, \rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.dimensionless_unscaled:
+            clabel = r"$\Delta^2_{21}$"
+        else:
+            raise ValueError(
+                "Accepted PS units: mK^2 Mpc^3, mK^2 Mpc^3/h^3, mK^2 or dimless."
```

Comment:
Again, I think coercing the units is nicer because some people will maybe have `K^2`